### DISCUSSION TITLE
Mamba F405MK2 4 motor fixed wing setup

### DISCUSSION THREAD

**Discussion by ArMaTeC** [iNavFlight/inav#7245] [Q&A]
Hi all,

I'm trying to set up a C130 fixed-wing I'm building but I'm running into an issue when trying to add all 4 motors.
If I run with just 2 motors set the ESC powers up and I get the normal beeps from the motors but as soon as I add a 3rd the start beep sounds but the all ok beeps do not.

Does anyone have any clues on why this would happen or could anyone let me know of a good setup for a cheap 4 motor fixed-wing INAV setup.

Firmware: INAV/MAMBAF405 3.0.1 Jul  4 2021 / 13:55:23 (753c4479)

Any help welcome

**Comment by stronnag** [comment]
The hardware / firmware does not support that configuration. [This wiki page](https://github.com/iNavFlight/inav/wiki/Boards,-Targets-and-PWM-allocations) may assist you in selecting a board that meets your specific requirement (4 FW motors and probably 4 servos), a non-typical / non-trivial requirement.

**Comment by ArMaTeC** [comment]
So from what I can see only the  KAKUTE F7 Mini V3 is the only one that support 4 fixed wing setups but that only supports a flying wing.
Unless there is anyway I can work around this I think I'm going to need to scrap this project for now or go down the pi route.

**Comment by stronnag** [comment]
There are probably a number of FCs that *could* support your use case (if you described it in terms of what you actually need in for number of motors and servos), some with a trivial code enhancement.

**Reply by ArMaTeC** [reply]
So I need it to have 5 servos and 4 motors

**Comment by stronnag** [comment]
If you want the FC to be able to control all of those, the MATEKH743 would be a option. It would need a small code adjustment; it provides 11 PWM outputs, with sufficient timer flexibility to meet your requirement.
(note the wiki only lists the first 8 outputs for simplicity / author laziness).

**Reply by ArMaTeC** [reply]
Ordered the flight controller I'll let ya know how I get on thank you

**Reply by stronnag** [reply]
ping me if you need help with getting it working. I'll be interested in seeing this.

**Reply by ArMaTeC** [reply]
@stronnag I have the flight controller now how would I go about assigning the 3/4 motor to the ports?

**Comment by stronnag** [comment]
It is necessary to redefine some outputs in `target.c` and rebuild the firmware. Note that for each distinct timer, (`TIMx`), there should be assigned *only* motor or servo, not a mix. An example is attached; 6 motors / 6 servos. Other combinations are possible, but this seems the most logical. 

I've also built firmware from this `target.c`; also attached. Note it's from the `master` (development`) branch (3.1.0), as this has the most recent H743 code. You can find a contemporary 3.1.0 configurator from http://seyrsnys.myzen.co.uk/inav-configurator-next/.

Unpack the zip file, use the configurator  `Load firmware [local]` option to load the hex file.

[inav_3.1.0_MATEKH743_M6S6.zip](https://github.com/iNavFlight/inav/files/6816935/inav_3.1.0_MATEKH743_M6S6.zip)

**Reply by ArMaTeC** [reply]
All 4 motors are working as required I will take some time to print out a new electronics box as this FC is a tad bigger and then ill get to the nitty-gritty of the servos.

Thank you kind sir you are a legend

**Reply by ArMaTeC** [reply]
@stronnag is there any way the change you did the targets.c would cause the onboard SDcard slot to stop working as the option is not there anymore and only shows serial

**Comment by stronnag** [comment]
SDcard is not yet working on MatekH743. 
This is planned to be rectified prior to the final 3.1 release.
